erpnext/patches/v12_0/set_serial_no_status.py

```diff
@@ -0,0 +1,11 @@
+from __future__ import unicode_literals
+import frappe
+
+def execute():
+	frappe.reload_doc('stock', 'doctype', 'serial_no')
+
+	for serial_no in frappe.db.sql("""select name from `tabSerial No`""", as_dict = 1):
+		doc = frappe.get_doc("Serial No", serial_no.get("name"))
+		if not doc.status:
+			doc.set_status()
```

Comment:
The number of serial nos can be huge and `get_doc` can take too much time.

Implement (copy) the logic of set_status here to avoid `get_doc` and calling `set_status`

Reply:
Done